lib/openstudio-standards/btap/geometry.rb

```diff
@@ -500,6 +500,80 @@ def self.filter_spaces_by_space_types(model, spaces_array, spacetype_array)
         end
         return returnarray
       end
+
+      ##
+      # Fetch a space's full height.
+      #
+      # @param space [OpenStudio::Model::Space] a space
+      #
+      # @return [Float] full height of space (0 if invalid input)
+      def self.space_height(space = nil)
+        return 0 unless space.is_a?(OpenStudio::Model::Space)
+
+        minZ = 1000
+        maxZ = 0
+
+        # An alternative to BTAP's "get_max_space_height_for_space_type".
+        #   - considering all surface types: "Floor" vs "Wall" vs "RoofCeiling"
+        #   - no presumption that atria floor are necessarily at ground level
+        space.surfaces.each do |surface|
+          minZ = [surface.vertices.min_by(&:z).z, minZ].min
+          maxZ = [surface.vertices.max_by(&:z).z, maxZ].max
+        end
+
+        maxZ < minZ ? 0 : maxZ - minZ
+      end
+
+      ##
+      # Fetch a space's width.
+      #
+      # @param space [OpenStudio::Model::Space] a space
+      #
+      # @return [Float] width of a space (0 if invalid input)
+      def self.space_width(space = nil)
+        return 0 unless space.is_a?(OpenStudio::Model::Space)
+
+        floors = TBD.facets(space, "all", "Floor")
+        return 0 if floors.empty?
+
+        # Automatically determining a space's "width" is not straightforward:
+        #   - a space may hold multiple floor surfaces at various Z-axis levels
+        #   - a space may hold multiple floor surfaces, with unique "widths"
+        #   - a floor surface may expand/contract (in "width") along its length.
```

Comment:
An attempt to systematically determine a space's _width_, to support BTAP in (better) managing NECB space type variants. The usual suspects are _corridors_, which are often spaghetti-shaped and vary in width, e.g. **SmallHotel**:

<img width="1645" alt="hotel" src="https://github.com/user-attachments/assets/3839f514-9f5f-4679-8177-895a6df6f058" />